### PR TITLE
change ci with cache(part2 of #1174)

### PR DESCRIPTION
**Pull request by huang12zheng**
## Changes

_Please list issues fixed by this PR here, using format "Fixes #the-issue-number"._

## Checklist

- [ ] An issue to be fixed by this PR is listed above.
- [ ] New tests are added to ensure new features are working. End-to-end tests are usually in the `./frb_example/pure_dart` example, more specifically, `rust/src/api.rs` and `dart/lib/main.dart`.
- [ ] The code generator is run and the code is formatted (via `just precommit`).
- [ ] If this PR adds/changes features, documentations (in the `./book` folder) are updated.
- [ ] CI is passing.

## Remark for PR creator

- Justfile is a task runner for the command line interface that allows you to run shell commands from a file named `justfile` in your project directory. You can use Justfile after [installing it](https://github.com/casey/just). Note that commands written in `justfile` of this repository are expected to be run in `bash`, not `cmd` or `powershell`. Running `just ...` commands in `cmd` or `powershell` will produce errors as the syntax is not compatible. On Windows, you can use `git bash` if you have Git installed.
- If fzyzcjy does not reply for a few days, maybe he just did not see it, so please ping him.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -302,6 +294,7 @@ jobs:
 
       - run: just install_ffigen
 
+      # use action.yaml due to I don't know what was done with the `Install crates`
```
Comment (fzyzcjy) [inline .github/workflows/ci.yaml:297]: uber-nit: Not very sure what this line of comment points to

File: .github/actions/setup/action.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+name: Setup Rust Builder
```
Comment (fzyzcjy) [inline .github/actions/setup/action.yaml:1]: Not very sure about the file naming. Seems this is just "do cargo cache", thus the name may be "cargo_cache" or something like that?

File: .github/actions/setup/action.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        path: |
+          ~/.cargo/bin/
+          ~/.cargo/registry/index/
+          ~/.cargo/registry/cache/
+          ~/.cargo/git/db/
+          ./frb_example/with_flutter/rust/target
+        key: ${{ runner.os }}-cargo-integrate-ios-${{ hashFiles('**/Cargo.lock') }}-${{ inputs.cache-key }}
+        restore-keys: |
+            ${{ runner.os }}-cargo-integrate-ios-${{ hashFiles('**/Cargo.lock') }}
+            ${{ runner.os }}-cargo-integrate-ios
```
Comment (fzyzcjy) [inline .github/actions/setup/action.yaml:24]: uber-nit: maybe no longer "integrate-ios", but should be something that is passed in as an argument?

**Review by fzyzcjy — COMMENTED** [review]
Good job! Only two nits and ready to merge

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (huang12zheng) [reply .github/workflows/ci.yaml:297]: - name: Install crates
  if: steps.test-cache.outputs.cache-hit != 'true'
  run: |
    sh -c "$(curl https://rustwasm.github.io/wasm-pack/installer/init.sh -sSf)" '' -f

Comment (fzyzcjy) [reply .github/workflows/ci.yaml:297]: Then I guess maybe we should not add this as a comment line?

File: .github/actions/setup/action.yaml (hunk 3 above)
Comment (fzyzcjy) [reply .github/actions/setup/action.yaml:24]: Not sure. If cache for different platforms will collide, then we should add this

**Comment by fzyzcjy** [comment]
not very sure, for what target?

File: .github/actions/setup/action.yaml (hunk 3 above)
Comment (huang12zheng) [reply .github/actions/setup/action.yaml:24]: 1. We do have arg -> ${{ runner.os }}
2. no fail happen for now. 
---
But you have a point
Curious. Could it have been needed before?
For now, I'm thinking about the fact that between different IOS, the previous product of rust, is the same with name,and would be overwritten
android in particular

Comment (huang12zheng) [reply .github/actions/setup/action.yaml]: * android output
[2023-04-12T16:48:44Z INFO  cargo_ndk::cli] /Users/runner/work/flutter_rust_bridge/flutter_rust_bridge/target/x86_64-linux-android/debug/libflutter_rust_bridge_example.so -> ../android/app/src/main/jniLibs/x86_64/libflutter_rust_bridge_example.so

Comment (huang12zheng) [reply .github/actions/setup/action.yaml]: * ios
[INFO  cargo_lipo::lipo] Creating universal library for flutter_rust_bridge_example

**Comment by fzyzcjy** [comment]
Good job!

**Pull request merged**